src/overlays/actors/ovl_En_Tru/z_en_tru.c

```diff
@@ -25,91 +89,1189 @@ const ActorInit En_Tru_InitVars = {
+            func_80A86924(this, 0);
+            func_80A86460(this);
+        } else if (gSaveContext.weekEventReg[16] & 0x10) {
+            func_80A86924(this, 6);
+        }
+
+        func_8013AED4(&this->unk_34E, 0, 7);
+        this->unk_34E &= ~(0x1000 | 0x8);
+        this->unk_34E |= 0x10;
+        this->actor.shape.rot.y = this->actor.world.rot.y;
+        this->actor.flags &= ~0x100;
+        this->unk_1E8 = 0;
+        this->actionFunc = func_80A87FD0;
+    }
+}
+
+void EnTru_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnTru* this = THIS;
+
+    if ((gSaveContext.entranceIndex != 0xC200) || (gSaveContext.weekEventReg[12] & 8)) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    }
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 24.0f);
+    SkelAnime_InitSV(globalCtx, &this->skelAnime, &D_0601AA60, NULL, this->jointTable, this->morphTable, 27);
+    Collider_InitAndSetSphere(globalCtx, &this->collider, &this->actor, &sSphereInit);
+    CollisionCheck_SetInfo2(&this->actor.colChkInfo, DamageTable_Get(0x16), &sColChkInfoInit);
+    this->unk_37C = -1;
+    func_80A86924(this, 0);
+    this->path = func_8013BEDC(globalCtx, this->actor.params & 0xFF, 255, &this->unk_384);
+    if (this->path != NULL) {
+        this->unk_384 = 1;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86384.s")
+    this->actor.targetMode = 0;
+    Actor_SetScale(&this->actor, 0.008f);
+    this->unk_34E = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86460.s")
+    if (gSaveContext.weekEventReg[16] & 0x10) {
+        func_80A86924(this, 5);
+    } else {
+        this->unk_388 = 0;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86674.s")
+    this->actionFunc = func_80A87FD0;
+    Actor_UpdateBgCheckInfo(globalCtx, &this->actor, 0.0f, 0.0f, 0.0f, 4);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86770.s")
+void EnTru_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnTru* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A868F8.s")
+    Collider_DestroySphere(globalCtx, &this->collider);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86924.s")
+void EnTru_Update(Actor* thisx, GlobalContext* globalCtx) {
+    EnTru* this = THIS;
+    f32 radius;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A8697C.s")
+    func_80A872AC(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A869DC.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86B0C.s")
+    func_80A868F8(this);
+    func_80A86B0C(this, globalCtx);
+    func_80A86DB8(this);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Tru/func_80A86BAC.s")
+    radius = this->collider.dim.worldSphere.radius + 30;
+    this->unk_388 = !(this->unk_34E & 0x80) ? 0 : 0;
```

Comment:
Both true and false cases evaluates to 0?

Reply:
Yeah.